x-pack/solutions/security/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/action_response_outputs.tsx

```diff
@@ -0,0 +1,214 @@
+
+    const {
+      canWriteFileOperations,
+      canReadActionsLogManagement,
+      canAccessEndpointActionsLogManagement,
+    } = useUserPrivileges().endpointPrivileges;
+
+    const hasMultipleAgents = useMemo(() => {
+      return action.agents.length > 1;
+    }, [action.agents.length]);
+
+    const consoleCommandName = useMemo(() => {
+      return RESPONSE_ACTION_API_COMMAND_TO_CONSOLE_COMMAND_MAP[action.command];
+    }, [action.command]);
+
+    return useMemo(() => {
+      return (
+        <div data-test-subj={getTestId()}>
+          {action.agents.map((agentId) => {
+            const { wasSuccessful, isCompleted, completedAt } = action.agentState[agentId];
+            const hostStatusMessage = !isCompleted
+              ? OUTPUT_MESSAGES.isPending(consoleCommandName)
+              : wasSuccessful
+              ? OUTPUT_MESSAGES.wasSuccessful(consoleCommandName)
+              : action.isExpired
+              ? OUTPUT_MESSAGES.hasExpired(consoleCommandName)
+              : OUTPUT_MESSAGES.hasFailed(consoleCommandName);
+            const hostName = action.hosts[agentId]?.name ?? agentId;
+            let hostOutput: React.ReactNode = null;
+
+            if (isCompleted && wasSuccessful) {
+              if (isGetFileAction(action)) {
+                hostOutput = (
+                  <ResponseActionFileDownloadLink
+                    action={action}
+                    agentId={agentId}
+                    canAccessFileDownloadLink={canWriteFileOperations}
+                    textSize="xs"
+                    data-test-subj={getTestId('getFileDownloadLink')}
+                  />
+                );
+              }
+
+              if (isExecuteAction(action)) {
+                hostOutput = (
+                  <ExecuteActionHostResponse
+                    action={action}
+                    agentId={agentId}
+                    canAccessFileDownloadLink={
+                      canAccessEndpointActionsLogManagement || canReadActionsLogManagement
+                    }
+                    textSize="xs"
+                    data-test-subj={getTestId('actionsLogTray')}
+                  />
+                );
+              }
+
+              if (isUploadAction(action)) {
+                hostOutput = (
+                  <EndpointUploadActionResult
+                    action={action}
+                    agentId={agentId}
+                    data-test-subj={getTestId('uploadOutput')}
+                    textSize="xs"
+                  />
+                );
+              }
+
+              if (isProcessesAction(action)) {
+                hostOutput = (
+                  <RunningProcessesActionResults
+                    action={action}
+                    agentId={agentId}
+                    data-test-subj="processesOutput"
+                    textSize="xs"
+                  />
+                );
+              }
+
+              if (isRunScriptAction(action)) {
```

Comment:
FYI: I've created a new [RunScriptoutput](https://github.com/elastic/kibana/pull/242441/files#diff-e795f5df6ff889799c7ebbf2f4a1578f5a0f064139a36ca026c5f3e5acca7150R104) component within my PR instead of reusing ExecuteHost. I'll probably merge after this PR so I'll make adjustments.